File: Java Script - Introduction to Programming/9.1. Problems for Champions - Part I/01. Crossing sequences.js
function crossingSequences([...arg]) {
    let [tribonacciFirst, tribonacciSecond, tribonacciThird, spiralCurrent, spiralStep] = [...arg.map(Number)]
    let tribonacciNumbers = [tribonacciFirst, tribonacciSecond, tribonacciThird]

    let tribonacciCurrent = tribonacciThird
    while (tribonacciCurrent < 1000000) {
        tribonacciCurrent = tribonacciFirst + tribonacciSecond + tribonacciThird

        tribonacciNumbers.push(tribonacciCurrent)
        tribonacciFirst = tribonacciSecond
        tribonacciSecond = tribonacciThird
        tribonacciThird = tribonacciCurrent
    }

    let spiralNumbers = [spiralCurrent]
    let spiralCount = 0
        spiralStepMul = 1

    while (spiralCurrent < 1000000) {
        spiralCurrent += spiralStep * spiralStepMul
        spiralNumbers.push(spiralCurrent)
        spiralCount++
        if (spiralCount % 2 == 0) {
            spiralStepMul++
        }
    }

    let found = false;
    for (let i = 0; i < tribonacciNumbers.length; i++) {
        for (let j = 0; j < spiralNumbers.length; j++) {
            if (tribonacciNumbers[i] == spiralNumbers[j] && tribonacciNumbers[i] <= 1000000) {
                console.log(tribonacciNumbers[i])
                found = true
                break
            }
        }
        if (found) break
    }

    if (!found) console.log("No")
}

// crossingSequences([1, 2, 3, 5, 2])
// crossingSequences([13, 25, 99, 5, 2])

File: Pyrhon - Introduction to Programming/9.1. Problems for Champions - Part I/01. Crossing sequences.py
tribonacci_first, tribonacci_second, tribonacci_third, spiral_current, spiral_step = [int(input()) for _ in range(5)]
tribonacci_numbers = [tribonacci_first, tribonacci_second, tribonacci_third]

tribonacci_current = tribonacci_third

while tribonacci_current < 1000000:
    tribonacci_current = tribonacci_first + tribonacci_second + tribonacci_third

    tribonacci_numbers.append(tribonacci_current)
    tribonacci_first = tribonacci_second
    tribonacci_second = tribonacci_third
    tribonacci_third = tribonacci_current


spiral_numbers = [spiral_current]
spiralCount = 0
spiralStepMul = 1

while spiral_current < 1000000:
    spiral_current += spiral_step * spiralStepMul
    spiral_numbers.append(spiral_current)
    spiralCount+= 1
    if spiralCount % 2 == 0:
        spiralStepMul+= 1


found = False
for i in range(0, len(tribonacci_numbers)):
    for j in range(0, len(spiral_numbers)):
        if tribonacci_numbers[i] == spiral_numbers[j] and tribonacci_numbers[i] <= 1000000:
            print(tribonacci_numbers[i])
            found = True
            break

        if found: break


if not found:
    print('No')


'''
1
2
3
5
2
#37
'''
